pkg/filesystem/resources.go

```diff
@@ -43,3 +43,17 @@ func LibexecPath() (string, error) {
 	// Compute the expected libexec path.
 	return filepath.Clean(filepath.Join(executablePath, "..", "..", "libexec")), nil
 }
+
+func resolveLink(executablePath string) (string, error) {
```

Comment:
Instead of pulling this into a different function, can we keep the original conditional structure?  Perhaps something like:

```
if target, err := os.Readlink(executablePath); err != nil {
    return "", fmt.Errorf("unable to read executable symbolic link target: %w", err)
} else if filepath.IsAbs(target) {
    executablePath = target
} else if resolved, err := filepath.Abs(filepath.Join(filepath.Dir(executablePath), target)); err != nil {
    return "", fmt.Errorf("unable to resolve executable symbolic link target: %w", err)
} else {
    executablePath = resolved
}
```

(Basically just adding the additional `else if filepath.IsAbs(target)` branch to keep the diff smaller and the code a bit more compact.)

Reply:
Thank you for the feedback! I've fixed the issue now with another conditional branch.

Previously I've used the additional function in order to declutter the conditions and make it more clear, how the logic behind this exactly work.